src/Symfony/Component/Cache/Traits/RedisTrait.php

```diff
@@ -174,28 +174,24 @@ public static function createConnection(#[\SensitiveParameter] string $dsn, arra
 
         $params += $query + $options + self::$defaultConnectionOptions;
 
-        if (isset($params['redis_sentinel']) && isset($params['sentinel_master'])) {
```

Comment:
This removes the fact that you cannot use those options together. You know silently ignore `redis_sentinel` when `sentinel_master` (or the new `sentinel`) is set. Doesn't this introduce a potential WTF debugging moment if `$query` and `$options` both configure this using different names for instance ?

Reply:
I don't think the exception is useful so I still prefer removing it.
I improved the merging logic to account for this case.